py-rse/configuration.Rmd

```diff
@@ -51,162 +53,361 @@ fonts:
 - Serif
 ```
 
-And here's an extension to our script template
-that uses `argparse` and provide
-a default configuration that the user can override:
+## Matplotlib configuration {#py-rse-configuration-matplotlib}
+
+To see overlay configuration in action,
+let's consider one of the most common tasks in data science --
+changing the size of the labels on a plot.
+The size of the labels on our Jane Eyre plot
+is fine for viewing on our screen,
+but they'll need to be bigger
+if we want to include the figure in a powerpoint presentation or report.
+
+```{r py-rse-configuration-jane-eyre-default, echo=FALSE, fig.cap="Word frequency distribution for the book Jane Eyre with default label sizes."}
+knitr::include_graphics("figures/py-rse-configuration/jane_eyre_default.png")
+```
+
+In order to configure the label sizes,
+we could use any of the overlay options described above:
+
+-  Edit the system-wide matplotlib configuration YAML file known as the `matplotlibrc` file
+-  Create a user-specific matplotlib style sheet (which also uses the YAML format) 
+-  Create a job-specific YAML file that the `plotcounts.py` script can parse
+-  Add some new command-line options to `plotcounts.py` 
+
+Let's consider these options one at a time.
+
+### Edit the matplotlibrc file
+
+Our first option would be to edit the system-wide matplotlib runtime configuration file
+(a.k.a. the `matplotlibrc` file).
+When we import matplotlib,
+it locates this YAML file in order to set the default characteristics
+of the plot (i.e. colors, sizes, etc). 
+We can find its location on our system by running the following command:
 
 ```python
-import argparse
-import yaml
+import matplotlib as mpl
+mpl.matplotlib_fname()
+```
 
-DEFAULT_CONFIG = {
-    'source': 'DEFAULT',
-    'logfile': '/tmp/log.txt',
-    'quiet': False,
-    'overwrite': False,
-    'fonts': ['Verdana', 'Serif']
-}
+```text
+/Users/amira/opt/anaconda3/lib/python3.7/site-packages/matplotlib/mpl-data/matplotlibrc
```

Comment:
Worth a note to deconstruct this path: the first part is where Anaconda is installed, `site-packages` is where packages go, and the rest is matplotlib's.